Makefile

```diff
@@ -3,14 +3,34 @@ clean:
 	rm -f dist/*.whl
 	rm -rf build
 
-build:
+# PyPI archive preparation
+build_pypi:
 	python setup.py sdist
+	python -m cython -a streaming_form_data/_parser.pyx -o annotation.html
 
-upload: build
+upload: build_pypi
 	twine upload dist/*
 
-test:
+# build the module and install to the system
+install:
+	python -m pip install . --ignore-installed
+
+# build the module in the current directory
+# it will be available for import from the project root directory
+build_local:
+	python setup.py build_ext -i
+
+# run tests and check code formatting
+test: build_local
 	PYTHONPATH=. py.test
 	flake8
 
-.PHONY: clean build upload
+# gather library function call statistics (time, count, ...)
+profile: build_local
+	PYTHONPATH=. python utils/benchmark.py -f tests/data/image-high-res.jpg -c image/jpeg
+
+# calculate library bandwidth
+speedtest: build_local
+	PYTHONPATH=. python utils/speedtest.py
```

Comment:
Do we need `PYTHONPATH` definition here?

Reply:
For development, what I do is install the package locally in development mode (`pip install -e .`), after which `PYTHONPATH` variable is not needed.